### PR TITLE
Clarify usage of spanmetrics connector with a prometheus remote write

### PR DESCRIPTION
**Pull request by ptodev**
This has been a source of confusion for several customers.

### CODE REVIEW

File: docs/sources/flow/reference/components/otelcol.connector.spanmetrics.md
Comment (ptodev) [inline docs/sources/flow/reference/components/otelcol.connector.spanmetrics.md:283]: This PR is notable for using these `collapse` features inside the Examples section of Flow component docs. AFAIK we haven't done this before. I think it could really help illustrate this issue to people who are not familiar with OTLP.

Screenshot 2024-01-05 at 18 00 45

File: docs/sources/flow/reference/components/otelcol.connector.spanmetrics.md
Comment (thampiotr) [inline docs/sources/flow/reference/components/otelcol.connector.spanmetrics.md:625]: Thanks for writing this up and coming up with solutions to this problem. I worry that for users this all seems complex and the Agent is doing a poor job in simplifying this for the end user. 

Ideally, I would like to have a brief description of the problem and the ability to choose between the two recommended solutions using a one-line of config. This could be a module or a component that can hide this complexity away from the user. 

Can we consider putting this problem/solutions description in a different document as a short-term way to unblock users, but open an issue to build something with a better user experience?

Comment (ptodev) [reply docs/sources/flow/reference/components/otelcol.connector.spanmetrics.md:625]: If there is another module/component which can abstract away these things, then that'd be great. However, this is the reference page for `otelcol.connector.spanmetrics`, so it feels like the right place to explain why it behaves the way it does and how to use it. 

I agree that we can try to be more straight to the point in this example section. I could make a new section outside of the Examples section where I explain the tendency to preserve resource attributes? And I could link to it from the Examples section, for users who want a deeper dive into why things are the way they are?

File: docs/sources/flow/reference/components/otelcol.connector.spanmetrics.md
Comment (ptodev) [reply docs/sources/flow/reference/components/otelcol.connector.spanmetrics.md:282]: As mentioned in the [other comment](https://github.com/grafana/agent/pull/5980/files#r1444833913), I could move the explanation about resource attributes to another section in this doc? IMO this reference doc should the the most detailed doc about this component and should explain in detail how it works. The resource attributes explanation does describe "what the component will do". 

Btw the first time I heard about the way this components threats resource attributes, I thought it was a bug 😆 I wanted to add this explanation to illustrate why it's not.

File: docs/sources/flow/reference/components/otelcol.connector.spanmetrics.md
Comment (ptodev) [reply docs/sources/flow/reference/components/otelcol.connector.spanmetrics.md]: Yes, both requests and errors are tracked using a metric with the same name - `calls`.  I agree with you that the phrasing here could be improved! I'll change it to this:

>  Requests are tracked using a `calls` metric with a `status.code` datapoint attribute set to `Ok`:

> Errors are tracked using a `calls` metric with a `status.code` datapoint attribute set to `Error`:

Let me know if you'd like me to change it further

File: docs/sources/flow/reference/components/otelcol.connector.spanmetrics.md
Comment (thampiotr) [reply docs/sources/flow/reference/components/otelcol.connector.spanmetrics.md:625]: > it feels like the right place to explain why it behaves the way it does

yeah, that certainly :) I just have doubts if this is the right place for what looks like a bit of a workaround... as mentioned, I'd prefer we had a way to address this with a single line of config, where we just say which of the two strategies to address the problem should be used. In practice it will be a few lines, of course, but we should perhaps consider a module or a change to the component in some way that will allow to resolve this problem in a more user-friendly way. 

Is there a way we can make this task simpler for the users?

Comment (ptodev) [reply docs/sources/flow/reference/components/otelcol.connector.spanmetrics.md]: We could make it simpler for users if we offer a module which contains three components:
1. A transform processor which removes unnecessary resource attributes
1. A spanmetrics connector
1. An otelcol.prometheus.exporter

I could raise an issue for such a module to be created? This sort of thing is why modules are created, so I suppose we should be able to do it. But it'd be better wait and to do it with Modules v2 rather than with v1.

Comment (thampiotr) [reply docs/sources/flow/reference/components/otelcol.connector.spanmetrics.md]: Yeah, I agree with the module approach 👍 
Migration to a new module style shouldn't be too hard, but we can prioritise this independently nevertheless.

File: docs/sources/flow/reference/components/otelcol.connector.spanmetrics.md
Comment (thampiotr) [inline docs/sources/flow/reference/components/otelcol.connector.spanmetrics.md:543]: Should we link at the bottom to `Sending metrics via a Prometheus remote write` saying that there's a way to ensure all samples are received?

**Review by thampiotr — APPROVED** [review]
Looks more clean now I believe :)

File: docs/sources/flow/reference/components/otelcol.connector.spanmetrics.md
Comment (ptodev) [reply docs/sources/flow/reference/components/otelcol.connector.spanmetrics.md:543]: It is not necessary, because using `otelcol.connector.spanmetrics` only with other OTLP-native components is totally ok. There is only a problem when converting the OTLP metrics to Prometheus format.

This "Handling of resource attributes" section simply describes the OTLP features.

**Comment by ptodev** [comment]
Hi @clayton-cornell - I intended to merge this one without your approval, while you were away :) Since your back, would you like to take a look? If you're busy I could merge it anyway. I think we will need to backport this to the 0.39 release too, once it's available.

File: docs/sources/flow/reference/components/otelcol.connector.spanmetrics.md
Comment (clayton-cornell) [inline docs/sources/flow/reference/components/otelcol.connector.spanmetrics.md:422]:
```suggestion
1. To preserve the values of all resource attributes, `otelcol.connector.spanmetrics` will produce two resource metrics.
```

**Review by clayton-cornell — COMMENTED** [review]
One quick tweak. Otherwise, lets get this merged and revisit once it's live (the content is ok as-is... it could use a cleanup, but that applies to the whole topic, not just this input).

**Review by clayton-cornell — APPROVED** [review]

**Pull request merged**
